test/framework/src/main/java/org/elasticsearch/transport/MockTcpTransport.java

```diff
@@ -294,26 +333,23 @@ public void onFailure(Exception e) {
                 @Override
                 protected void doRun() throws Exception {
                     StreamInput input = new InputStreamStreamInput(new BufferedInputStream(activeChannel.getInputStream()));
+                    // as per interruption policy of CancellableThreads we don't need to check for interrupts.
```

Comment:
Ok, then I revert the change because I think it's best we just exit the loop in case we're interrupted.